src/m365/spo/commands/applicationcustomizer/applicationcustomizer-set.ts

```diff
@@ -0,0 +1,190 @@
+}
+
+class SpoApplicationCustomizerSetCommand extends SpoCommand {
+  private readonly allowedScopes: string[] = ['Site', 'Web', 'All'];
+
+  public get name(): string {
+    return commands.APPLICATIONCUSTOMIZER_SET;
+  }
+
+  public get description(): string {
+    return 'Updates an existing Application Customizer on a site';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --webUrl <webUrl>'
+      },
+      {
+        option: '-t, --title [title]'
+      },
+      {
+        option: '-i, --id [id]'
+      },
+      {
+        option: '-c, --clientSideComponentId [clientSideComponentId]'
+      },
+      {
+        option: '--newTitle [newTitle]'
+      },
+      {
+        option: '-p, --clientSideComponentProperties [clientSideComponentProperties]'
+      },
+      {
+        option: '-s, --scope [scope]', autocomplete: this.allowedScopes
+      }
+    );
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        title: typeof args.options.title !== 'undefined',
+        id: typeof args.options.id !== 'undefined',
+        clientSideComponentId: typeof args.options.clientSideComponentId !== 'undefined',
+        newTitle: typeof args.options.newTitle !== 'undefined',
+        clientSideComponentProperties: typeof args.options.clientSideComponentProperties !== 'undefined',
+        scope: typeof args.options.scope !== 'undefined'
+      });
+    });
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.id && !validation.isValidGuid(args.options.id)) {
+          return `${args.options.id} is not a valid GUID`;
+        }
+
+        if (args.options.clientSideComponentId && !validation.isValidGuid(args.options.clientSideComponentId)) {
+          return `${args.options.clientSideComponentId} is not a valid GUID`;
+        }
+
+        if (args.options.scope && this.allowedScopes.indexOf(args.options.scope) === -1) {
+          return `'${args.options.scope}' is not a valid application customizer scope. Allowed values are: ${this.allowedScopes.join(',')}`;
+        }
+
+        if (!args.options.newTitle && !args.options.clientSideComponentProperties) {
+          return `Please specify an option to be updated`;
+        }
+
+        return validation.isValidSharePointUrl(args.options.webUrl);
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      { options: ['id', 'title', 'clientSideComponentId'] }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const appCustomizer = await this.getAppCustomizerToUpdate(logger, args.options);
+      await this.updateAppCustomizer(logger, args, appCustomizer);
```

Comment:
Slight change: let's just pass the options in, instead of the entire args object. I'll do this one merging

Reply:
Done!